crates/sui-core/src/authority_aggregator.rs

```diff
@@ -1873,6 +2038,20 @@ where
     }
 }
 
+fn aggregate_errors(
+    errors: &Vec<(SuiError, Vec<AuthorityName>, StakeUnit)>,
+) -> BTreeMap<String, (Vec<AuthorityName>, StakeUnit)> {
```

Comment:
Thinking about how I would implement the detection of  `TransactionInputsObjects` errors, it would be helpful (less brittle), if instead of this map being keyed by `String`, it was keyed by `SuiError`.  The type supports `Hash` and `Eq` so this would be feasible if we returned a `HashMap` instead of a `BTreeMap` here.  Is there a good reason to keep the order of this map deterministic?

Otherwise, I'm tempted to say we should just keep the errors as a `Vec<(SuiError, Vec<AuthorityName>, StakeUnit)>`.

Reply:
Ordering doesn't matter.

In terms of `String` or `SuiError`: `SuiError` has many variants that are tuple/struct like, I'm not sure how to group them given the inner value could be different. But this is a fair point - using `String` is cumbersome when we categorize them in QuorumDriver, We could either: use `SuiError` with default inner value or use `String` which makes it a bit less convenient for QuorumDriver. wdyt?

Reply:
If it's possible that the grouping logic is specific to each class of error, I think it might make sense to not attempt to group them, and just return the `Vec`?

Reply:
We need a way to look at (ErrorType, StakeUnit) though

Reply:
I suppose my suggestion is to leave the aggregation logic up to downstream users who can decide at what granularity they want to group `ErrorType`s up.  I know what we need to do for the transaction input object errors following your change, but like you said, there are lots of different kinds of error, and I'm not sure the same technique applies to all of them, at least when it comes to figuring out whether they are retryable or not.

Reply:
@amnn  As i'm writing the downstream logic (basically, `fn convert_to_error_if_nonretryable`), i realize aggregated errors are actually very awkward to consume. So, like you suggested, I'm going to pass down a vector of errors down the line and handle the processing in quorum driver. Will update the PR